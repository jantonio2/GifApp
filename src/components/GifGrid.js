import React, { useEffect, useState } from 'react'

export const GifGrid = ({category}) => {

  const [count, setCount] = useState(0);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async() => {
    const url = 'https://api.giphy.com/v1/gifs/search?q=Game+of+Thrones&limit=10&api_key=G8Eh9rUQomhjc30iU4SQJ660FaIVfPgB'
    const resp = await fetch(url);
    const {data} = await resp.json();

    const gifs = data.map(img => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url
      }
    })
    

    console.log(gifs);
  }

  return (
    <div>
      <h3>{category}</h3>
      <h2>{count}</h2>
      <button onClick = {() => setCount(count + 1)}></button>
    </div>
  )
}
