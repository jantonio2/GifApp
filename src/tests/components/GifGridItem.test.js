import { shallow } from 'enzyme';
import { GifGridItem } from './../../components/GifGridItem';

describe('Prueba en el componente GifGridItem', () => {

  const title = "Game of Thrones";
  const url = "https://localhost";

  test('debe de mostrar el componente correctamente', () => {
    const wrapper = shallow(
      <GifGridItem 
        title = {title}
        url = {url}
      />);

    expect(wrapper).toMatchSnapshot();
  });
});
