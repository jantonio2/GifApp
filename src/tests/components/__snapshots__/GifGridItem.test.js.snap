// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el componente GifGridItem debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Game of Thrones"
    src="https://localhost"
  />
  <p>
    Game of Thrones
  </p>
</div>
`;
